core/src/session.rs

```diff
@@ -67,7 +67,9 @@ impl Session {
         credentials: Credentials,
         cache: Option<Cache>,
     ) -> Result<Session, SessionError> {
-        let ap = apresolve(config.proxy.as_ref(), config.ap_port).await;
+        let ap = try_apresolve(config.proxy.as_ref(), config.ap_port)
+            .await
+            .expect("Could not resolve access point");
```

Comment:
No way.

This function has already `Result` as return type, no need to panic.

Reply:
Oh right. Good thing you're looking over my shoulder.